components/messenger.rst

```diff
@@ -0,0 +1,194 @@
+.. index::
+   single: Messenger
+   single: Components; Messenger
+
+The Messenger Component
+=======================
+
+    The Messenger component helps application send and receive messages to/from other applications or via message queues.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/messenger
+
+Alternatively, you can clone the `<https://github.com/symfony/messenger>`_ repository.
+
+.. include:: /components/require_autoload.rst.inc
+
+Concepts
+--------
+
+.. image:: /_images/components/messenger/overview.png
+
+**Sender**:
+   Responsible for serializing and sending the message to _something_. This
+   something can be a message broker or a third party API for example.
+
+**Receiver**:
+   Responsible for deserializing and forwarding the messages to handler(s). This
+   can be a message queue puller or an API endpoint for example.
+
+**Handler**:
+   Given a received message, contains the user business logic related to the
+   message. In practice, that is just a PHP callable.
+
+Bus
+---
+
+The bus is used to dispatch messages. MessageBus' behavior is in its ordered
+middleware stack. When using the message bus with Symfony's FrameworkBundle, the
+following middlewares are configured for you:
+
+#. ``LoggingMiddleware`` (logs the processing of your messages)
+#. ``SendMessageMiddleware`` (enables asynchronous processing)
+#. ``HandleMessageMiddleware`` (calls the registered handle)
+
+Example::
+
+  use Symfony\Component\Messenger\MessageBus;
+  use Symfony\Component\Messenger\HandlerLocator;
+  use Symfony\Component\Messenger\Middleware\HandleMessageMiddleware;
+
+    use App\Message\MyMessage;
```

Comment:
```diff
-  use Symfony\Component\Messenger\MessageBus;
-  use Symfony\Component\Messenger\HandlerLocator;
-  use Symfony\Component\Messenger\Middleware\HandleMessageMiddleware;
-
-    use App\Message\MyMessage;
+    use App\Message\MyMessage;
+    use Symfony\Component\Messenger\MessageBus;
+    use Symfony\Component\Messenger\HandlerLocator;
+    use Symfony\Component\Messenger\Middleware\HandleMessageMiddleware;
```

Reply:
I'm not sure how it even ended like that 😂 Changed!